### PR TITLE
issue #19 check new gem version periodically

### PR DESCRIPTION
**Pull request by laurent-martin**
implemented this new small feature

### CODE REVIEW

**Review by gegles — APPROVED** [review]
That's great, thanks @laurent-martin!

File: README.md [hunk 1]
```diff
@@ -1532,6 +1543,7 @@ OPTIONS:
         --secret=VALUE               access key secret for node
         --secrets=VALUE              access key secret for node
         --test-mode=ENUM             skip user validation in wizard mode: yes, no
+        --version-check-days=VALUE   period to check neew version in days (zero to disable)
```
Comment (dwosk) [inline README.md:1546]: @laurent-martin typo

File: lib/aspera/cli/plugins/config.rb [hunk 2]
```diff
@@ -106,8 +108,10 @@ def initialize(env,tool_name,help_url,version)
           self.options.add_opt_simple(:secret,"access key secret for node")
           self.options.add_opt_simple(:secrets,"access key secret for node")
           self.options.add_opt_boolean(:test_mode,"skip user validation in wizard mode")
+          self.options.add_opt_simple(:version_check_days,Integer,"period to check neew version in days (zero to disable)")
```
Comment (dwosk) [inline lib/aspera/cli/plugins/config.rb:111]: typo

File: lib/aspera/cli/plugins/config.rb [hunk 3]
```diff
@@ -122,6 +126,48 @@ def get_secrets
...
+            end
+            current_date=Date.today
+            Log.log.info("last check: #{last_check_date.class}")
+            if last_check_date.nil? or (current_date - last_check_date) > delay_days
+              Log.log.info("days elapsed #{last_check_date.is_a?(Date) ? current_date - last_check_date : last_check_date.class.name}")
+              last_check_array[0]=current_date.strftime("%Y/%m/%d")
+              check_date_persist.save
+              check_data=check_gem_version
+              if check_data[:need_update]
+                Log.log.warn("A new version is available: #{check_data[:latest]}. You have #{check_data[:current]}. Upgrade with: gem update #{check_data[:nname]}")
```
Comment (dwosk) [inline lib/aspera/cli/plugins/config.rb:165]: should be `#{check_data[:name]}`

**Pull request merged**